### DISCUSSION TITLE
Do we need to call initAuth() in all handlers?

### DISCUSSION THREAD

**Discussion by lazycipher** [gladly-team/next-firebase-auth#708] [Q&A]
Should we call the initAuth() in all handlers or it should be fine without that?

**Comment by kmjennison** [comment]
Yes, depending on your backend architecture. If, like on Vercel, they're deployed as standalone services, you'll need to.

**Reply by lazycipher** [reply]
gotcha, thanks a lot for clearing it out.

@kmjennison, can you help me with something.

I've a function `getData(abc)`, which uses firebase admin. I am importing this function in a component's `getServerSideProps` like following:

```jsx
export async function getServerSideProps(context) {
 const response = await getData(abc);
return{{}}
}

```

it's throwing error 

```
FirebaseAppError: The default Firebase app does not exist. Make sure you call initializeApp() before using any of the Firebase services.

```

Now, my question is, where should I call the initAuth() function? 
- in the component/page, 
- in the getServerSideProps
- in the file containing getData()
- in the function getData()

I will be really grateful if you may help me out.

**Reply by kmjennison** [reply]
You should init in the file containing your endpoint handler, outside of functions. Check out the example app in this repository if you haven't already.
